xero-app-store.yaml

```diff
@@ -91,6 +91,139 @@ paths:
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/UsageRecord"
+              example: '{
+                "usageRecordId": "2a90f7d9-4c10-4a23-9c34-b47225167a48",
+                "subscriptionId": "391be708-878a-4741-acfb-c0bc9a165bdb",
+                "subscriptionItemId": "b22b150f-a0db-447d-9117-d922110add64",
+                "quantity": 22,
+                "testMode": true,
+                "timeStamp": "2022-09-13T02:11:22Z"
+              }'
+        "404":
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetails"
+          description: "When a failure occurs in the endpoint"
+      security:
+        - OAuth2: [marketplace.billing]
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/UpdateUsageRecord"
+            example: '{
+              "quantity": 10
+            }'
+        description: Contains the quantity for the usage record to update
+        required: true
+  "/subscriptions/{subscriptionId}/usage-records":
+    get:
+      tags:
+        - AppStore
+      operationId: getUsageRecords
+      summary: "Gets all usage records related to the subscription"
+      parameters:
+        - $ref: "#/components/parameters/subscriptionId"
+      responses:
+        "200":
+          description: Success - return a list of all usage record submitted against this subscription for this subscription period
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/UsageRecordsList"
+              example: '{
+                  "usageRecords": [
+                      {
+                          "usageRecordId": "206100d3-e18b-4c59-9372-e98e4367a73f",
+                          "subscriptionId": "7c7684cf-bbfa-4fa8-8f44-eba8d1acdc5d",
+                          "subscriptionItemId": "264f6fa2-65e6-48f4-8a92-57cde5499742",
+                          "productId": "b040838b-f85e-4e7c-8dba-5a8501e5d312",
+                          "pricePerUnit": 2.0,
+                          "quantity": 12,
+                          "testMode": false,
+                          "timeStamp": "2022-01-01T00:00:00Z"
+                      },
+                      {
+                          "usageRecordId": "47d61dc9-3c99-4587-8d55-0985f47df4a6",
+                          "subscriptionId": "7c7684cf-bbfa-4fa8-8f44-eba8d1acdc5d",
+                          "subscriptionItemId": "264f6fa2-65e6-48f4-8a92-57cde5499742",
+                          "productId": "ec8033e8-5e3c-4795-bfb4-bfb85fa4557e",
+                          "pricePerUnit": 5.0,
+                          "quantity": 2,
+                          "testMode": false,
+                          "timeStamp": "1970-01-01T00:00:00Z"
+                      }
```

Comment:
probably too pedantic. Even though this is just an example, could it really happen that you would have two records that would be sent years apart? In this case  "timeStamp": "2022-01-01T00:00:00Z" and  "timeStamp": "1970-01-01T00:00:00Z"